main.go

```diff
@@ -333,11 +335,35 @@ func checkEmptyArgs(args ...string) bool {
 }
 
 // choose http or https
-func serve(logger log.Logger, tls bool, port, key, cert string) {
+func serve(logger log.Logger, tlsEnabled bool, port, key, certPath string) {
 	portStr := fmt.Sprintf(":%v", port)
-	if tls {
+	if tlsEnabled {
+		chain, err := tls.LoadX509KeyPair(certPath, key)
+		if err != nil {
+			logger.Log("err", "failed to load TLS certificate or private key")
+			os.Exit(1)
+		}
+
+		cert, err := x509.ParseCertificate(chain.Certificate[0]) // Leaf is always the first entry
+		if err != nil {
+			logger.Log("err", "error parsing TLS certificate")
+			os.Exit(1)
+		}
+
+		if _, err := cert.Verify(x509.VerifyOptions{}); err != nil {
```

Comment:
The documentation for `Verify` says 
> If opts.Roots is nil and system roots are unavailable the returned error will be of type SystemRootsError.

Are you sure you're not going to run into the error by using an empty `VerifyOptions` struct?

Otherwise looks good.